engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -209,24 +211,60 @@ class NodeCompiler(definitions: ProcessDefinition[ObjectWithMethodDef],
   }
 
   def compileProcessor(n: Processor, ctx: ValidationContext)
-                     (implicit nodeId: NodeId): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
-    compileService(n.service, ctx, None)
+                     (implicit nodeId: NodeId, metaData: MetaData): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
+    compileService(n.service, ctx, None, None)
   }
 
-  def compileEnricher(n: Enricher, ctx: ValidationContext, outputVar: Option[OutputVar]) (implicit nodeId: NodeId): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
-    compileService(n.service, ctx, outputVar)
+  def compileEnricher(n: Enricher, ctx: ValidationContext, outputVar: Option[OutputVar])
+                     (implicit nodeId: NodeId, metaData: MetaData): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
+    compileService(n.service, ctx, outputVar, None)
   }
 
-  private def compileService(n: ServiceRef, validationContext: ValidationContext, outputVar: Option[OutputVar] = None)(implicit nodeId: NodeId): NodeCompilationResult[compiledgraph.service.ServiceRef] = {
+  def compileService(n: ServiceRef,
+                     validationContext: ValidationContext,
+                     outputVar: Option[OutputVar],
+                     serviceCollectorOpt: Option[ServiceInvocationCollectorForContext])
```

Comment:
I have some refactoring of ServiceQuery where it will be used: https://github.com/TouK/nussknacker/pull/1406/commits/424dcd76179d113bf674bef8a583f39df82471b2#diff-d9fceaf5a5e622a7b0668f2c619b3a8c796290ceeef80fea9830fdbbd0db4bbb
Didn't include it in this PR to keep size reasonable :)